pkg/agents/team5/messages.go

```diff
@@ -206,3 +206,91 @@ func (a *CustomAgent5) HandleStateIntendedFoodTaken(msg messages.StateIntendedFo
 	a.updateIntentionFoodMemory(msg.SenderID(), statement)
 	//a.Log("New value of intendedFood", infra.Fields{"statement": statement, "memory": a.socialMemory[msg.SenderID()].intentionFood})
 }
+
+func (a *CustomAgent5) RejectTreaty(msg messages.ProposeTreatyMessage) {
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), false)
+	a.SendMessage(reply)
+	a.Log("Rejected treaty", infra.Fields{"proposerID": msg.SenderID(), "proposerFloor": msg.SenderFloor(), "treatyID": msg.TreatyID()})
+}
+
+func (a *CustomAgent5) HandleProposeTreaty(msg messages.ProposeTreatyMessage) {
+	treaty := msg.Treaty()
+	if len(a.ActiveTreaties()) > 0 || treaty.Request() == messages.Inform {
+		//TODO: Have a checkForConflicts function that determines if we can feasibly sign up to multiple treaties.
+		//For now, only agree to sign up to 1 treaty at a time for simplicity
+		//Also, reject any inform treaties as I'm not sure what they are or how to handle them
+		a.RejectTreaty(msg)
+		return
+	}
+
+	if (treaty.Request() == messages.LeaveAmountFood && treaty.RequestOp() == messages.EQ) || treaty.RequestOp() == messages.LE || treaty.RequestOp() == messages.LT {
+		//Reject all treaties that ask you to leave less food, don't see why you would do this
+		//Reject any treaty that asks you to leave a specific amount of food as this would lead to multiple people just not eating as only 1 agent can eat and fufill that criteria
+		a.RejectTreaty(msg)
+		return
+	}
+
+	if (treaty.ConditionOp() == messages.LE || treaty.ConditionOp() == messages.LT) && (treaty.Condition() == messages.HP || treaty.Condition() == messages.AvailableFood) {
+		//Reject treaties that have a condition of being less than something
+		// Why agree to treaty which is bounded by you having lower HP
+		// Why agree to treaty which is bounded by the amount of food on platform being low
+		// All of these are conditions where you will be desperate for food so shouldn't be limitting yourself if you do get the opurrtunity to eat
+		a.RejectTreaty(msg)
+		return
+	}
+
+	if (treaty.ConditionOp() == messages.GE || treaty.ConditionOp() == messages.GT) && treaty.Condition() == messages.Floor {
```

Comment:
with GE and GT, doesn't this currently reject treaties that are bounded by you being higher in the tower? I think this might be the wrong way around

Reply:
I wasn't sure how to interpret the floor values in terms of GT or LT. As being higher up in the tower means having a floor value less than (LT) those below you. So if the condition is GE or GT than it means being lower down in the tower as the floor value you have will be larger. Not sure if that makes sense?